### PR TITLE
apparmor.InstallDefault(): fix order of remove/close and fix linting warnings

### PR DESCRIPTION
**Pull request by thaJeztah**
splitting this from https://github.com/moby/moby/pull/42303

Current versions of gosec showed warnings for these:

    contrib/apparmor/main.go:49:2: G307: Deferring unsafe method "Close" on type "*os.File" (gosec)
        defer f.Close()
        ^
    profiles/apparmor/apparmor.go:101:2: G307: Deferring unsafe method "Close" on type "*os.File" (gosec)
        defer f.Close()
        ^
    profiles/apparmor/apparmor.go:118:2: G307: Deferring unsafe method "Close" on type "*os.File" (gosec)
        defer file.Close()
        ^

Looking at the existing code: https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/profiles/apparmor/apparmor.go#L99-L102

It looks like these will be executed in the incorrect order; defer statements
are executed in _reverse_ order in which they're added. For example:

    package main

    import "fmt"

    func main() {
        defer fmt.Println("executing f.Close()")
        defer fmt.Println("executing os.Remove(profilePath)")
    }

Will print:

    executing os.Remove(profilePath)
    executing f.Close()

Grouping them together in a single defer makes sure they're executed in the expected order, and makes it slightly less error-prone.

(those defers were added in https://github.com/moby/moby/commit/2f7596aaef3a9f8ec1f2d0937462d9263bee8b6b / https://github.com/moby/moby/pull/26518)

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
@cpuguy83 @AkihiroSuda @cyphar ptal (looks like those defers were added in https://github.com/moby/moby/pull/26518)

File: contrib/apparmor/main.go [hunk 1]
```diff
@@ -46,7 +46,9 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
-	defer f.Close()
+	defer func() {
+		_ = f.Close()
+	}()
```
Comment (thaJeztah) [inline contrib/apparmor/main.go:51]: Also happy to add a `//nolint` instead for these simple cases

File: profiles/apparmor/apparmor.go [hunk 2]
```diff
@@ -98,8 +98,10 @@ func InstallDefault(name string) error {
 	}
 	profilePath := f.Name()
 
-	defer f.Close()
-	defer os.Remove(profilePath)
```
Comment (cyphar) [inline profiles/apparmor/apparmor.go:102]: The order of these statements doesn't matter (at least on Linux). You can remove a file that is still open.

**Review by cyphar — COMMENTED** [review]
I don't have a strong opinion on this, but I don't understand why this is a lint warning at all -- what is "unsafe" about calling `f.Close()`. Is it just because we don't check the return value (if so, there's a lot of Go code that is going to start getting lint errors -- `defer f.Close()` is used in so many projects and in standard library examples).

**Comment by thaJeztah** [comment]
Arf, forgot I had this one.

> The order of these statements doesn't matter (at least on Linux). You can remove a file that is still open.

Yes, you're absolutely right. I guess I was focusing on the order, and that didn't come to mind when looking at that.

> what is "unsafe" about calling f.Close(). Is it just because we don't check the return value

I started to look at that as well, because when updating the linter I got a ton more of those messages, but it looks like it's just similar to the `errCheck` linter, but different linter or different warning. I adjusted the linter config in another PR (https://github.com/moby/moby/pull/42445), which takes care of these.


Anyway; thanks for looking! I'm gonna close this one as it's not needed.

**Pull request closed without merging**